packages/cra-template-mephisto-review/template/src/GridRenderer.jsx

```diff
@@ -1,22 +1,38 @@
 import React, { useState } from "react";
 import { Redirect } from "react-router-dom";
 import { useMephistoReview } from "mephisto-review-hook";
-import { H5, H3, H2, H1 } from "@blueprintjs/core";
+import { H4, H3, H2, H1, InputGroup } from "@blueprintjs/core";
+import { Tooltip2 } from "@blueprintjs/popover2";
 import GridView from "./GridView";
 import Pagination from "./components/Pagination";
 import "./GridRenderer.css";
 
 function GridRenderer() {
   const resultsPerPage = 9;
   const [page, setPage] = useState(1);
+  const [filters, setFilters] = useState("");
+  const [filterTimeout, setFilterTimeout] = useState(null);
+
   const {
     data,
     isFinished,
     isLoading,
     error,
     mode,
     totalPages,
-  } = useMephistoReview({ page, resultsPerPage });
+  } = useMephistoReview({ page, resultsPerPage, filters });
+
+  const delaySetFilters = (filtersStr) => {
+    if (filterTimeout) {
+      clearTimeout(filterTimeout);
+    }
+
+    setFilterTimeout(
+      setTimeout(() => {
+        setFilters(filtersStr);
+      }, 2000)
```

Comment:
Waiting 2 seconds is a decent approach to ensure we're not spamming the backend with requests that won't be used, however we should probably have a manual trigger too? Could use Enter to submit the given search w/o waiting.